sentry-core/src/main/java/io/sentry/core/EnvelopeSender.java

```diff
@@ -179,11 +186,19 @@ private void processEnvelope(final @NotNull SentryEnvelope envelope, final @Null
                     SentryLevel.WARNING,
                     "Timed out waiting for item submission: %s",
                     session.getSessionId());
+
+                if (hint instanceof Retryable) {
```

Comment:
Should we pull the `if (hint instanceof Flushable) {` logic into some function and reuse it? The fact we need to change it multiple times is a bad sign

Reply:
makes sense, needs to be public though :P u need to give up of one of them

Reply:
I guess we just make it public then and add that `@InternalApi` annotation. or a `io.sentry.internal` package which we can document (or don't even need to) that we'll break without bumping a major version